### PR TITLE
update README

### PR DESCRIPTION
**Pull request by Seif-Mamdouh**
## Changes
This PR introduces several enhancements and clarifications to improve our bytes documentation and workflow:

- Added a note clarifying that projects don't need to be overly large or complex.
- Included a new section detailing the process for creating GitHub issues.
GitHub Labels Enhancement:
   - Introduced a "brainstorm" label for GitHub issues.
   - Added an example issue for reference: Issue #2 
- Added Implemented a `.gitignore` file to better manage version control.
- Incorporated a high-quality animated GIF showcasing how bytes are displayed on coreyja.com.
- corrected the `test.yaml` file to ensure proper CI/CD functionality.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -1,52 +1,51 @@
...
-    }
-}
\ No newline at end of file
+name: Rust
+
+on:
+  push:
+    branches: ["main"]
+    paths:
+      - '**/cargo/**'
```
Comment (Seif-Mamdouh) [inline .github/workflows/test.yaml:7]: rust tests should run on any cargo folder in the directory

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -1,52 +1,47 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Build
+        run: cargo build --verbose --no-default-features
+      - name: Run tests
+        run: cargo test --verbose --no-default-features --lib
```
Comment (coreyja) [inline .github/workflows/test.yaml]: Lets remove the `--no-default-features` from both and the `--lib` from the tests!

I doubt we will have people toggle feature flags, but if they do we should run tests for them!

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -1,52 +1,47 @@
...
-    if content.is_empty() {
-        println!("No tasks found.");
-    } else {
-        for (index, task) in content.lines().enumerate() {
-            println!("{}: {}", index, task);
-        }
-    }
-}
\ No newline at end of file
+name: Rust
```
Comment (coreyja) [inline .github/workflows/test.yaml:1]: Nice!

To get this workflow to actually run for each sub-dir, I think we want to make this whole repo a Cargo Workspace!
https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html

Which is how most of our repos have been structured!

Basically we just need to add a `Cargo.toml` to the root of the repo with a block like the following to tell Cargo where to look for 'sub-projects'

```toml
[workspace]

members = [
    "byte0",
]
```

File: .gitignore
Comment (coreyja) [inline .gitignore:1]: Lets add some contents here! A good starting place might be the default one Github has for Rust
https://github.com/github/gitignore/blob/main/Rust.gitignore

The only thing we will want to change from ^^ is to REMOVE the `Cargo.lock` line! We _want_ to commit our lock files!

Here's a post I wrote last year about lockfiles, just to share lol
https://coreyja.com/posts/weekly/20230901/

File: README.md [hunk 4]
```diff
@@ -4,45 +4,68 @@ Welcome to Open Source Bytes, a collection of fun and educational Rust games! 
...
 Play the Bytes on [here](https://coreyja.com/bytes)!
 
-#### TL:DR
+#### 📚TL:DR
 - Bytes are Rust coding games designed for pure Rust learning! 🚀
 - We love Rust and want everyone else to learn. What better way to learn than by reading Rust code and guessing what bugs there are? 🐛
 - We want you to contribute your own Bytes to make the Rust challenges even more fun! 🤝 Be creative! 
 
-[![Watch the video](https://img.youtube.com/vi/aiTte9mmPGQ/0.jpg)](https://www.youtube.com/watch?v=aiTte9mmPGQ&t=209s)
+![Bytes Demo](bytes-demo.gif)
```
Comment (coreyja) [inline README.md:12]: Love this!

**Comment by coreyja** [comment]
Looking great! 

Two small nit-ish things!

Looks like there is still a `byte0/target` directory in here, lets clean that up before merging!

And then _really_ nitpicky 😆 The `cargo.toml` at the top level is in lowercase, and I'm used to `Cargo.toml`!
Doesn't look like the tool cares but can we capitalize it?

After those two are in I think we are probably good to merge this and then we can move the repo under my name and get ready to post on Reddit maybe Monday next week? Or Friday isn't crazy if you think people will do this on weekends?
I tend to do Monday things, but not super sure about the student crowd?

**Pull request merged**